packages/ckeditor5-utils/src/focustracker.ts

```diff
@@ -111,28 +242,161 @@ export default class FocusTracker extends /* #__PURE__ */ DomEmitterMixin( /* #_
 	 */
 	public destroy(): void {
 		this.stopListening();
+
+		this._elements.clear();
+		this._externalViews.clear();
+
+		this.isFocused = false;
+		this.focusedElement = null;
 	}
 
 	/**
-	 * Stores currently focused element and set {@link #isFocused} as `true`.
+	 * Stores currently focused element as {@link #focusedElement} and sets {@link #isFocused} `true`.
 	 */
 	private _focus( element: Element ): void {
-		clearTimeout( this._nextEventLoopTimeout! );
+		// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": _focus() on element`, element );
+
+		this._clearBlurTimeout();
 
 		this.focusedElement = element;
 		this.isFocused = true;
 	}
 
 	/**
-	 * Clears currently focused element and set {@link #isFocused} as `false`.
-	 * This method uses `setTimeout` to change order of fires `blur` and `focus` events.
+	 * Clears currently {@link #focusedElement} and sets {@link #isFocused} `false`.
+	 *
+	 * This method uses `setTimeout()` to change order of fires `blur` and `focus` events ensuring that moving focus between
+	 * two elements within a single focus tracker's scope, will not cause `[ blurA, focusB ]` sequence but just `[ focusB ]`.
+	 * The former would cause a momentary change of `#isFocused` to `false` which is not desired because any logic listening to
+	 * a focus tracker state would experience UI flashes and glitches as the user focus travels across the UI.
 	 */
 	private _blur(): void {
-		clearTimeout( this._nextEventLoopTimeout! );
+		const isAnyElementFocused = this.elements.find( element => element.contains( document.activeElement ) );
+
+		// Avoid blurs originating from external FTs when the focus still remains in one of the #elements.
+		if ( isAnyElementFocused ) {
+			return;
+		}
+
+		const isAnyExternalViewFocused = this.externalViews.find( view => {
+			// Do not consider external views's focus trackers as focused if there's a blur timeout pending.
+			return view.focusTracker.isFocused && !view.focusTracker._blurTimeout;
```

Comment:
Can we change `focusTracker._blurTimeout` to a public getter like `focusTracker.isBlurPending`?

Reply:
Why would you make this public? I think there's no reason for anyone to interact with this already complex and fragile piece of code.

Reply:
It could be `FocusTracker#_blurTimeout` too (public+internal), it just doesn't feel good that you reach to other FT instance private member here.

Reply:
Let's leave it as it is per https://www.typescriptlang.org/docs/handbook/2/classes.html#cross-instance-private-access